modules/perception/obstacle/camera/cipv/cipv.cpp

```diff
@@ -0,0 +1,773 @@
+/******************************************************************************
+ * Copyright 2018 The Apollo Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *****************************************************************************/
+
+#include <math.h>
+#include "modules/perception/obstacle/camera/cipv/cipv.h"
+#include "modules/common/log.h"
+
+namespace apollo {
+namespace perception {
+namespace obstacle {
+
+Cipv::Cipv(void) {}
+
+Cipv::~Cipv(void) {
+  //    clear();
+}
+
+bool Cipv::init() {
+  _b_image_based_cipv = false;
+  _debug_level = 0;  // 0: no debug message
+                     // 1: minimal output
+                     // 2: some important output
+                     // 3: verbose message
+                     // 4: visualization
+                     // 5: all
+                     // -x: specific debugging, where x is the specific number
+  _time_unit = AVERAGE_FRATE_RATE;
```

Comment:
format like this needs to be changed later

Reply:
Thank you for reviewing it! I will definitely change the format ASAP. Please let me know what is the best way to do.